### DISCUSSION TITLE
Release process current state and possible update

### DISCUSSION THREAD

**Discussion by OrlinVasilev** [vmware-tanzu/velero#6514] [General discussions]
Hi Community!
Based on our community meeting there is a question about release management and how we can improve if we want to improve!

please share your thoughts on what is working well and what needs improvement! 

Orlix

**Comment by mateusoliveira43** [comment]
Hi everyone

My idea would be to remove the need of [adding a changelog](https://velero.io/docs/v1.11/code-standards/#adding-a-changelog) during PRs using [Latest Changes tool](https://github.com/tiangolo/latest-changes). With it, after each PR, CI would commit the changelog info of that PR, already in desired format (using a template), removing the need of [make changelog](https://velero.io/docs/v1.11/release-instructions/#changelog-and-docs-pr).

### Examples of usage

- https://github.com/mateusoliveira43/cly/commits/main
- https://github.com/tiangolo/fastapi/commits/master

## Pros

- no need to manually write changelogs during PRs
- every PR would have a changelog
  - in my opinion, this is good because even small PRs would be referenced in Velero releases (for example, I think I already have 2 PRs merged, but both without changelog, so my name will not appear in future releases. I think this is good to attract more contributors). But we could also just trigger it using the label used today, for example

## Cons

- every PR merged will result in a second commit to the target branch (@sseago)
  -  the best scenario would be prior to merge, add the changelog to the commit, but I believe this is still not possible in GitHub actions today https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

## Related issues

- https://github.com/vmware-tanzu/velero/issues/3589
- https://github.com/vmware-tanzu/velero/issues/2522

@mpryc please also share your idea

**Comment by draghuram** [comment]
I don't have any new suggestions but it would be good to document release procedure in "Contribute" section. In particular, guidelines to back-porting changes in both main repo as well as plug-ins should be documented.

**Reply by OrlinVasilev** [reply]
it's available in the docs have seen this? https://velero.io/docs/main/release-instructions/

**Reply by draghuram** [reply]
Thanks. I checked it now and it does contain many details. But it doesn't seem to have the procedure for back-porting changes to released versions.
